libraries/apollo-mockserver/src/jsMain/kotlin/com/apollographql/apollo3/mockserver/MockServer.kt

```diff
@@ -48,18 +49,28 @@ actual class MockServer actual constructor(override val mockServerHandler: MockS
         socket.end()
       }
     }
-  }.listen()
+  }
+
+  init {
+    server.listen()
+  }
 
   override suspend fun url() = url ?: suspendCoroutine { cont ->
-    url = "http://localhost:${server.address().unsafeCast<AddressInfo>().port}/"
-    server.on("listening") { _ ->
+    val server = requireNotNull(this.server) { "Server is not listening, please call listen() before calling address()" }
```

Comment:
Can `this.server` ever be null?

Reply:
You're right, it can't, this is a bad copy/paste from v4, I'll remove it